ethsigner/src/main/java/tech/pegasys/ethsigner/requesthandler/sendtransaction/RetryMechanism.java

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright 2019 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+package tech.pegasys.ethsigner.requesthandler.sendtransaction;
+
+import java.io.IOException;
+
+import io.vertx.core.buffer.Buffer;
+import io.vertx.core.http.HttpClientResponse;
+
+public interface RetryMechanism<T> {
+
+  boolean mustRetry(final HttpClientResponse response, final Buffer body);
+
+  void retry(final T context, Runnable sender) throws IOException;
```

Comment:
Throwing an `IOException` seems odd for a retry handler, as surely some of the IOExceptions would trigger a retry attempt (perhaps not throw or wrap as something else?)

Reply:
Moved to runtime errors .... (or derivations thereof).